src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -512,25 +645,104 @@ sliceSepByMax predicate cnt (Fold fstep finitial fextract) =
 
     extract (Tuple' _ r) = fextract r
 
+-- | See 'Streamly.Internal.Data.Parser.escapedSliceSepBy'.
+--
+-- /Internal/
+--
+{-# INLINE escapedSliceSepBy #-}
+escapedSliceSepBy :: 
+    MonadCatch m => (a -> Bool) -> (a -> Bool) -> Fold m a b -> Parser m a b
+escapedSliceSepBy isSep isEsc (Fold fstep finitial fextract) =
+
+    Parser step initial extract
+
+    where
+    
+    initial = Tuple' Nothing <$> finitial
+
+    step (Tuple' maybePrevEscape s) a =
+        case maybePrevEscape of
+            Nothing ->
+                if isEsc a
+                then return $ Partial 0 (Tuple' (Just a) s)
+                else
+                    if isSep a
+                    then Done 0 <$> fextract s
+                    else
+                        do
+                            nextS <- fstep s a
+                            return $ Partial 0 (Tuple' Nothing nextS)
+            Just prevEsc ->
```

Comment:
The doc does not say anything about discarding the escape. You discard the escape if the succeeding element is an escape or a separator.
`\\` = `\`
`\,` = `,`
`\a` = `\a`
Assuming `\` is escape and `,` is sep.

Reply:
Please update the doc if this is the required behaviour.